environments/se/core/src/main/java/org/jboss/weld/environment/se/Weld.java

```diff
@@ -976,4 +984,40 @@ public boolean equals(Object obj) {
 
     }
 
+    private void validateMetadataAvailableAsBeanClass() {
+        if (!selectedAlternatives.isEmpty()) {
+            String result = checkMetadata.apply(selectedAlternatives);
+            if (result != null) {
+                throw WeldSELogger.LOG.metadataNotFoundInBeanClasses(Alternative.class.getSimpleName().toLowerCase(), result);
+            }
+        }
+
+        if (!enabledInterceptors.isEmpty()) {
+            String result = checkMetadata.apply(enabledInterceptors);
+            if (result != null) {
+                throw WeldSELogger.LOG.metadataNotFoundInBeanClasses(Interceptor.class.getSimpleName().toLowerCase(), result);
+            }
+        }
+
+        if (!enabledDecorators.isEmpty()) {
+            String result = checkMetadata.apply(enabledDecorators);
+            if (result != null) {
+                throw WeldSELogger.LOG.metadataNotFoundInBeanClasses(Decorator.class.getSimpleName().toLowerCase(), result);
+            }
+        }
+
+    }
+
+    private final Function<List<Metadata<String>>, String> checkMetadata = new Function<List<Metadata<String>>, String>() {
```

Comment:
There's no need to use function here - it could be a stateless private method, i.e. `private String checkMetadata(List<Metadata<String> metadata)`.

I'm missing a simple test :-)